### PR TITLE
dev-lang/rakudo: update to 2021.10 version

### PR DESCRIPTION
**Pull request by Miezhiko**

### CODE REVIEW

File: dev-lang/rakudo/rakudo-2021.10.ebuild [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+src_compile() {
+	emake NQP_JARS="${NQP}" BLD_NQP_JARS="${NQP}"
+}
+
+src_install() {
+	emake DESTDIR="${D}" NQP_JARS="${NQP}" BLD_NQP_JARS="${NQP}" install
+}
+
+src_test() {
```
Comment (thesamesam) [inline dev-lang/rakudo/rakudo-2021.10.ebuild:70]: After src_compile and before src_install to match the order in which it is executed.

File: dev-lang/moarvm/moarvm-2021.10.ebuild [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+LICENSE="Artistic-2"
+SLOT="0"
+
+IUSE="asan clang debug doc +jit optimize static-libs ubsan"
+
+RDEPEND="dev-libs/libatomic_ops
+		>=dev-libs/libuv-1.26:=
+		dev-libs/libffi:="
+DEPEND="${RDEPEND}
+	clang? ( >=sys-devel/clang-3.1 )
```
Comment (thesamesam) [inline dev-lang/moarvm/moarvm-2021.10.ebuild:30]: Clang should be in BDEPEND unless it links against libclang (does it?)

File: dev-lang/moarvm/moarvm-2021.10.ebuild [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	dev-lang/perl"
```
Comment (thesamesam) [inline dev-lang/moarvm/moarvm-2021.10.ebuild:31]: Perl certainly seems like at least a BDEPEND as it calls it in the ebuild.

File: dev-lang/moarvm/moarvm-2021.10.ebuild [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	einfo "--prefix '${prefix}'"
+	einfo "--libdir '${libdir}'"
+	local myconfigargs=(
+		"--prefix" "${prefix}"
+		"--has-libuv"
+		"--has-libatomic_ops"
+		"--has-libffi"
+		"--libdir" "${libdir}"
+		"--compiler" "$(usex clang clang gcc)"
+		"$(usex asan        --asan)"
```
Comment (thesamesam) [inline dev-lang/moarvm/moarvm-2021.10.ebuild:52]: I'd drop the whitespace. The devmanual is clear that it's preferred to not do this.

File: dev-lang/nqp/nqp-2021.10.ebuild [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+		dev-java/jline:0
+		dev-java/jna:4
+	)
+	moar? ( ~dev-lang/moarvm-${PV}[clang=] )"
+RDEPEND="${CDEPEND}
+	java? ( >=virtual/jre-1.8 )"
+DEPEND="${CDEPEND}"
+BDEPEND="${CDEPEND}
+	clang? ( sys-devel/clang )
+	java? ( >=virtual/jdk-1.8 )
```
Comment (thesamesam) [inline dev-lang/nqp/nqp-2021.10.ebuild:36]: I think we're supposed to use :* if it works with any Java version as far as we know.

(Also, note that nqp seems to get the BDEPEND part right.)

File: dev-lang/nqp/nqp-2021.10.ebuild [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+REQUIRED_USE="|| ( java moar )"
+
+CDEPEND="java? (
+		dev-java/asm:4
+		dev-java/jline:0
+		dev-java/jna:4
+	)
+	moar? ( ~dev-lang/moarvm-${PV}[clang=] )"
+RDEPEND="${CDEPEND}
+	java? ( >=virtual/jre-1.8 )"
```
Comment (thesamesam) [inline dev-lang/nqp/nqp-2021.10.ebuild:32]: :* again, probably.

**Comment by Miezhiko** [comment]
@thesamesam it seems like it was updated before review, we can maybe keep this open until the next update, I'm not sure

**Pull request closed without merging**